### DISCUSSION TITLE
Bot talent points

### DISCUSSION THREAD

**Discussion by bkmangx1** [trickerer/Trinity-Bots#370] [Q&A] [answered]
Hey, can you explain how bot talent points work? I thought it would be similar to players, but I have notice some odd things especially at low levels. For example a level 25 mage normally has 16 talent points, but bots are exceeding this number. If I set him to use fire spec, he has correct fire talents for that level, but is also using Icy Veins and Improved Blizzard. With /bonk it seems like bot has 16 points in every tree.

**Comment by trickerer** [comment, marked as answer]
Yes, bots gain talents in a different way than players.
1. Bots grab all talents available for they level in their main talent tree
2. Bots grab all talents available up to level 25 required (4th tier) in side talent trees
3. In many cases max talent rank is taken instead of a progression, so for example level 10 Warrior may gain +5% crit from Cruelty talent (I don't remember if this one is actually the case)

So, at lower levels bots have serious advantage over players in terms of... everything pretty much?

Why this happened? Spell rotations for bots are kinda universal and cover all specs, but they were written long ago and have many many small quirks which rely on having certain talents, so talent gain conditions were never changed.
